File: api/callback/definitions/callback.lua
---@class ClientCallbackProvider
---@field triggerServerCallback fun(name: string, cb: fun(result: any): nil, ...): nil
---@field awaitServerCallback fun(name: string, ...): any

---@class ServerCallbackProvider
---@field registerServerCallback fun(name: string, cb: fun(playerId: integer, ...): any): nil

File: api/callback/definitions/callback.ts
export interface ClientCallbackProvider {
    triggerServerCallback(name: string, cb: (result: any) => void, ...args: any[]): void;
    awaitServerCallback(name: string, ...args: any[]): any;
}

export interface ServerCallbackProvider {
    registerServerCallback(name: string, cb: (playerId: number, ...args: any[]) => any): void;
}
